filters/filters.go

```diff
@@ -96,6 +96,12 @@ type FilterContext interface {
 
 	// Allow filters to create their own spans
 	ParentSpan() opentracing.Span
+
+	// Returns a clone of the FilterContext including a brand new request object.
+	SplitWithRequest(cr *http.Request) (FilterContext, error)
```

Comment:
I thought the request cloning would happen in the context. What's the reason for not doing it there?

Reply:
To augment the request context from outside. I tried adding SplitWithOptions and SplitOptions to support overrides but I can't use SplitOptions struct from filters because it creates a circular import dependency.

Reply:
Rolled back to the Split signature.